serverless/src/main.rs

```diff
@@ -44,7 +67,76 @@ pub fn main() {
         stats_flusher,
     });
 
-    if let Err(e) = mini_agent.start_mini_agent() {
-        error!("Error when starting serverless trace mini agent: {e}");
+    tokio::spawn(async move {
+        let res = mini_agent.start_mini_agent().await;
+        if let Err(e) = res {
+            error!("Error when starting serverless trace mini agent: {e:?}");
+        }
+    });
+
+    let mut metrics_flusher = if dd_use_dogstatsd {
+        let metrics_aggr = Arc::new(Mutex::new(
+            MetricsAggregator::new(Vec::new(), CONTEXTS)
+                .expect("Failed to create metrics aggregator"),
+        ));
+
+        info!("Starting dogstatsd");
+        let _ = start_dogstatsd(dd_dogstatsd_port, &metrics_aggr).await;
+        info!("dogstatsd-udp: starting to listen on port {dd_dogstatsd_port}");
+
+        match dd_api_key {
+            Some(dd_api_key) => {
+                let metrics_flusher = Flusher::new(
+                    dd_api_key,
+                    Arc::clone(&metrics_aggr),
+                    build_fqdn_metrics(dd_site),
+                );
+                Some(metrics_flusher)
+            }
+            None => {
+                error!("DD_API_KEY not set, won't flush metrics");
+                None
+            }
+        }
+    } else {
+        info!("dogstatsd disabled");
+        None
+    };
```

Comment:
All of this could be abstracted into the `start_dogstatsd` method

Reply:
Refactored to move the metrics aggregator and flusher into `start_dogstatsd`

Reply:
https://github.com/DataDog/libdatadog/blob/55201cf0181104a17e7eefc13baa7804ab1de595/serverless/src/main.rs#L102-L143